### PR TITLE
{AKS} `az aks get-versions`: get-versions using newer backend api

### PR DESCRIPTION
**Pull request by xiangwei36**
**Related command**
az aks get-versions

**Description**
AKS offer a new simpler api for listKubernetesVersions, to replace legacy api ListOrchestratorProfile, which was designed to support docker swam and apache mesosphere.

**Testing Guide**
compare out put between old and new az aks get-versions, data should remain same, but json/table output have minor changes.

**History Notes**
[aks] BREAKING CHANGE: Both json payload and table format changed for `az aks get-versions`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by xiangwei36** [comment]
@FumingZhang, I am helping @haitch to submit this PR. Let me know if there is other issues.

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py [hunk 1]
```diff
@@ -477,7 +477,7 @@ def generate_ppg_id(self, resource_group, location):
     def _get_versions(self, location):
         """Return the previous and current Kubernetes minor release versions, such as ("1.11.6", "1.12.4")."""
         versions = self.cmd(
-            "az aks get-versions -l westus2 --query 'orchestrators[].orchestratorVersion'").get_output_in_json()
+            "az aks get-versions -l westus2 --query 'values[*].patchVersions.keys(@)[]'").get_output_in_json()
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py:480]: Is this change necessary? To me, the recording file is not changed, so I guess without the change, the test could pass.

**Review by FumingZhang — COMMENTED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py (hunk 1 above)
Comment (xiangwei36) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py:480]: The unit test that has _get_versions is test_aks_create_service_no_wait which is live_only.

Comment (xiangwei36) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py:480]: @haitch can you confirm?

Comment (haitch) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py:480]: @FumingZhang , I think I just did a search of "get-versions", and update here.

And, I am expecting this need to be updated as well, unless the hybrid test is something different.

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/hybrid_2020_09_01/test_aks_commands.py:480]: OK, as the recording file for the hybrid version is not updated, I thought we may keep the test case unchanged. But CI has passed, shouldn't block the change.

**Review by FumingZhang — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/_completers.py [hunk 2]
```diff
@@ -31,12 +31,12 @@ def get_k8s_versions_completion_list(cmd, prefix, namespace, **kwargs):  # pylin
...
     """Return a list of Kubernetes versions available for a new cluster."""
-    from azure.cli.command_modules.acs._client_factory import cf_container_services
+    from azure.cli.command_modules.acs._client_factory import cf_managed_clusters
     from jmespath import search
 
-    results = cf_container_services(cli_ctx).list_orchestrators(location, resource_type='managedClusters').as_dict()
+    results = cf_managed_clusters(cli_ctx).list_kubernetes_versions(location).as_dict()
     # Flatten all the "orchestrator_version" fields into one array
-    return search('orchestrators[*].orchestrator_version', results)
+    return search("values[*].patchVersions.keys(@)[]", results)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/acs/_completers.py:39]: May I ask if this modification will change the output logic and cause a breaking change to customers?

Comment (xiangwei36) [reply src/azure-cli/azure/cli/command_modules/acs/_completers.py:39]: This is not a breaking change. Customers will not experience any interruptions.

**Review by zhoxing-ms — APPROVED** [review]
